test/extended/operators/olm.go

```diff
@@ -71,3 +71,57 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
 		})
 	}
 })
+
+// This context will cover test case: OCP-23440, author: jiazha@redhat.com
+var _ = g.Describe("[Feature:Platform] the end user use OLM", func() {
+	defer g.GinkgoRecover()
+
+	var oc = exutil.NewCLI("olm-23440", exutil.KubeConfigPath())
+
+	var (
+		buildPruningBaseDir = exutil.FixturePath("testdata", "olm")
+		operatorGroup       = filepath.Join(buildPruningBaseDir, "operatorgroup.yaml")
+		catalogSourceConfig = filepath.Join(buildPruningBaseDir, "csc.yaml")
+		etcdSub             = filepath.Join(buildPruningBaseDir, "etcd-subscription.yaml")
+		etcdCluster         = filepath.Join(buildPruningBaseDir, "etcd-cluster.yaml")
+	)
+
+	files := []string{operatorGroup, catalogSourceConfig, etcdSub}
+	g.It(" subscribe the single etcd operator", func() {
+		g.By("Cluster-admin user subscribe the operator resource")
+		for _, v := range files {
+			configFile, err := oc.AsAdmin().Run("process").Args("--ignore-unknown-parameters=true", "-f", v, "-p", "NAME=test-operator", fmt.Sprintf("NAMESPACE=%s", oc.Namespace()), "MARKETPLACE=openshift-marketplace").OutputToFile("config.json")
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = oc.AsAdmin().WithoutNamespace().Run("create").Args("-f", configFile).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+		}
+		time.Sleep(90 * time.Second)
```

Comment:
Can we poll or watch instead of sleeping?

Reply:
Yeah, I also don't like to use `sleep`, but I couldn't find a better solution. If we use the `poll`, sometimes, the subscription failed to create, and it will hang there, and I'm not sure the exact rounds we should query.

Reply:
@ecordell Do you have any suggestions for this step? Thanks!

Reply:
@ecordell Done, updated it, thanks!